lib/plexy/config.ex

```diff
@@ -39,13 +41,38 @@ defmodule Plexy.Config do
   Used to gain access to the application env.
     ## Examples
 
+       iex> Application.put_env(:my_config, HerokuApi, heroku_api_url: "https://api.heroku.com")
+       iex> Plexy.Config.get(:my_config, [HerokuApi, :heroku_api_url])
+       "https://api.heroku.com"
+       iex> Plexy.Config.get(:my_config, [HerokuApi, :not_set], "and a default")
+       "and a default"
+
        iex> Application.put_env(:my_config, :redis_url, "redis://localhost:6379")
        iex> Plexy.Config.get(:my_config, :redis_url)
        "redis://localhost:6379"
        iex> Plexy.Config.get(:my_config, :foo, "and a default")
        "and a default"
   """
-  def get(config_name, key, default \\ nil) do
+  @spec get(atom(), atom() | [atom(), ...], any()) :: any()
+  def get(config_name, key, default \\ nil)
+
+  def get(config_name, [key | keys], default) do
+    default_resolver = fn
+      nil ->
+        default
+
+      found ->
+        found
+    end
+
+    config_name
+    |> Application.get_env(key)
+    |> get_in(keys)
+    |> default_resolver.()
```

Comment:
Is it worth having the convenient resolver function? Or is it simpler to just have something like:

```elixir
  config_name
    |> Application.get_env(key)
    |> get_in(keys) || default
    |> resolve(default)
```

Reply:
`|> get_in(keys) || default` is not valid syntax

also by using this resolver function we can match on `nil` which still allows users to set values to `false`

Reply:
its not valid syntax? And yes, false makes sense